e2e_test/batch/functions/array_cat.slt.part

```diff
@@ -0,0 +1,59 @@
+query T
+select array_cat(array[1,2,3], array[4,5,6]);
+----
+{1,2,3,4,5,6}
+
+query T
+select array[1,2,3] || array[4,5,6];
+----
+{1,2,3,4,5,6}
+
+query T
+select array_cat(array[1,2,3], 4);
```

Comment:
Just FYI, I have no strong opinion about how we do it: 🤪 

In PG `array_cat` doesn't support `array` + `elem`. It only supports `array` + `array` 

https://github.com/postgres/postgres/blob/90247e742f849794d061a0444071647728054b45/src/backend/utils/adt/array_userfuncs.c#L211-L216

And `||` = `array_append` + `array_prepend` + `array_cat`...

```sql
postgres=# select oprcode from pg_operator where oprname = '||';
     oprcode
-----------------
 array_append
 array_prepend
 array_cat
 textcat
 bitcat
 byteacat
 textanycat
 anytextcat
 tsvector_concat
 tsquery_or
 jsonb_concat
(11 rows)
```